src/app/backend/resource/pod/detail.go

```diff
@@ -179,6 +207,182 @@ func getPodController(client kubernetes.Interface, nsQuery *common.NamespaceQuer
 	return &ctrl, nil
 }
 
+func populateVolumeSourceInfo(volumes []v1.Volume, volumeName string) VolumeMountInfo {
+	for _, volume := range volumes {
+		if volume.Name == volumeName {
+			// yes, this is exponential, but N is VERY small, so the malloc for creating a named dictionary would probably take longer
+			return populateVolumeSourceInfoHelper(volume)
+		}
+	}
+	return VolumeMountInfo{}
+}
+
+// https://godoc.org/k8s.io/api/core/v1#Volumes
+func populateVolumeSourceInfoHelper(volume v1.Volume) VolumeMountInfo {
```

Comment:
huh... that is a lot. To be honest I'd probably rather pass `VolumeSource` to the frontend and extend our current `PersistentVolumeSource` object in the `backendapi.ts` to handle more sources, then map it directly in the frontend.

We are actually handling volume source in the `kd-persistent-volume-source` already so it could be made more generic.

Reply:
Do we really wanna do that ?
I mean, in the end of the day, I will have to add a bunch of entries in `backendapis.ts`, the backend would be simpler but the frontend would have the very same `populateVolumeSourceInfoHelper()` function, except that it will be written in `typescript` instead of `golang` BUT all these fiends that the frontend does not use at all would be sent to the frontend, which means a lot of useless data would be floating around for no reason.

Reply:
It is mostly handled in the frontend already and it is handled directly in the template. We could use generic component to display it.

https://github.com/kubernetes/dashboard/blob/master/src/app/frontend/resource/cluster/persistentvolume/detail/source/template.html

`backendapi` also already contains partially mapped volume source.
https://github.com/kubernetes/dashboard/blob/bd025c33bd36e214185d91796d12f6030b191c23/src/app/frontend/typings/backendapi.ts#L1170-L1181

Reply:
I know that.
I even started adding more parts of `backedapi.ts`:

https://github.com/kubernetes/dashboard/pull/5752/commits/3278c5b9a0c6a3466f61fdc6802c1dcec31cd3b0#diff-b29e54225c4ac5e7640f491d4d776b1d6410ca5fc4aafda232b1fb243ad89845

But in the end of the day:
* the structure is complex,
* it has multiple levels of nested elements
* some of them may change from time to time and keeping track of if may be hard (while on `golang`, the k8s library and the compiler does that for us automatically and for free
* Also, although it's a very complex structure, all we really need is a "name/id/folder path", i.e. a string.

But, we process this on the frontend, we will have to pass a huge complex structure though the network, only to parse a string afterwards. So... not very efficient.

Hence I ask you if you are really sure this is the path we should go.

Reply:
The thing is that we mostly have to provide this in the frontend to display all this info. We are already displaying a lot of info in the PV detail view. Every provider has a bit different schema and we can't really extract specific info. It has to be handled on a per-provider basis to properly display all data in the frontend.

Example
![Zrzut ekranu z 2021-01-21 13-47-15](https://user-images.githubusercontent.com/2285385/105353324-608b8600-5bef-11eb-9a15-e36ee639ad53.png)